File: problem_six/6.js
(function doMagic (max) {
  return console.log(square_of_sums(max) - sum_of_squares(max));
})(100);

function square_of_sums (max) {
  var sum = 0;
  for (var i=1; i<=max; i++){
    sum += i;
  }
  return Math.pow(sum, 2);
}

function sum_of_squares (max) {
  var sum = 0;
  for (var i=1; i<=max; i++){
    var square = Math.pow(i, 2);
    sum += square;
  }
  return sum;
}

File: problem_six/6.rb
def sum_of_squares(range)
  range.inject(0){|memo, number| memo += number**2}
end

def square_of_sums(range)
  range.inject(:+) ** 2
end

def do_magic(range)
  puts (square_of_sums(range) - sum_of_squares(range))
end

do_magic([*1..100])